server/controllers/UserControllers.js

```diff
@@ -0,0 +1,361 @@
+    return User
+    .create({
+      fullname: req.body.fullname,
+      username: req.body.username,
+      password,
+      email: req.body.email,
+      RoleId: req.body.RoleId,
+    })
+    .then((user) => {
+      const token = jwt.sign({
+        UserId: user.id,
+        userFullname: user.fullname,
+        userUsername: user.username,
+        userEmail: user.email,
+        userRoleId: user.RoleId,
+      }, secret, {
+        expiresIn: '72h'
+      });
+      return res.status(201).send({
+        message: 'User successfully created',
+        success: true,
+        user,
+        token,
+      });
+    })
+    .catch(() => res.status(400).send({
+      message: 'User credentials already exist',
+    }));
+  },
+/**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object containing the user's login details
+   */
+  login(req, res) {
+    return User
+      .findOne({
+        where: {
+          username: req.body.username
+        }
+      }).then((user) => {
+        if (!user) {
+          return res.status(400).send({ message: 'User not found' });
+        }
+        const passkey = bcrypt.compareSync(req.body.password, user.password);
+        if (passkey) {
+          const token = jwt.sign({
+            UserId: user.id,
+            userFullname: user.fullname,
+            userUsername: user.username,
+            userEmail: user.email,
+            userRoleId: user.RoleId,
+          }, secret, {
+            expiresIn: '24h'
+          });
+          return res.status(200).send({
+            status: 'ok',
+            success: true,
+            message: 'You are successfully logged in',
+            UserId: user.id,
+            token,
+          });
+        }
+        return res.status(400).send({ message: 'Incorrect password' });
+      })
+      .catch(error => res.send(error.message));
+  },
+
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object containing list of users with pagination
+   */
+  listUsers(req, res) {
+    const limit = req.query.limit || 10;
+    const offset = req.query.offset || 0;
+
+    if (isNaN(limit) || isNaN(offset)) {
+      return res.status(400).send({
+        message: 'limit and offset must be an integer'
+      });
+    }
+    return User.findAndCount({
+      limit,
+      offset,
+      attributes: ['id', 'fullname', 'username', 'email', 'RoleId'],
+    })
+    .then(user => res.status(200).send({
+      pagination: {
+        row: user.rows,
+        paginationDetails: pagination(user.count, limit, offset)
+      }
+    }))
+    .catch(error => res.status(403).send(error.message));
+  },
+
+
+  /**
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} returns a response object containing the list of user registered
+   */
+  searchUsers(req, res) {
+    let searchKey = '%%';
+    if (req.query.q) {
+      searchKey = `%${req.query.q}%`;
```

Comment:
This could be written using `iLike`. [More details here](http://docs.sequelizejs.com/manual/tutorial/querying.html#where)